tcwebhooks-web-ui/src/main/resources/buildServerResources/WebHook/js/editWebhook.js

```diff
@@ -339,23 +339,23 @@ function populateBuildHistoryAjax(locator) {
 			},
 			success: function (response) {
 				var myselect = $j('<select>');
-				myselect.append( $j('<option></option>').val(null).html("Choose a Build...") );
+				myselect.append( $j('<option></option>').val(null).text("Choose a Build...") );
```

Comment:
For some reason, this is no longer working either. The AJAX call is made and contains valid data, but the "Choose a Build..." message, and the build list are never populated.

![image](https://user-images.githubusercontent.com/88687/60383079-983c5580-9ac0-11e9-84e3-77494b23df52.png)

Reply:
Looking at the content, it looks like it's there, just not rendering.
```
<option value="">Choose a Build...</option>
<option value="3390">tcDummyDeb#2.0.3256 - SUCCESS (5 minutes ago)</option>
<option value="3389">tcDummy &lt;script&gt;alert(1);&lt;/script&gt;#2.0.3256 - SUCCESS (5 minutes ago)</option>
<option value="3388">tcDummyDeb#2.0.3255 - SUCCESS (15 minutes ago)</option>
```

Reply:
Versus the old code:
```
<option value="">Choose a Build...</option>
<option value="136461">tcDummyDeb &lt;script&gt;alert(1);&lt;/script&gt;#1.0.46647 - SUCCESS (2 minutes ago)</option>
<option value="136456">tcWebHooks - GitHub#387 - SUCCESS (a day ago)</option>
<option value="136453">Ace Editor#134 - SUCCESS (a day ago)</option>
<option value="136450">tcChatBot - Local#21 - SUCCESS (2 days ago)</option><option value="136449">tcChatBot - Local#20 - SUCCESS (2 days ago)</option>
```

Reply:
I'm not sure what the issue is here. I copied and pasted the whole function into a new blank page, added the scripts (jquery & moment) that are in the repo, added a few elements and some dummy data, and it works fine as-is. Maybe step through it..?

Reply:
Fyi. I am using the jQuery bundled with TeamCity if it's a recent version of TeamCity. So the one in the repo is old. TeamCity 2019.1.1 appears to be using jQuery 1.12.

Reply:
Is the select or anything *surrounding* the select changing somehow? A styling issue? I find it strange that the same HTML result would result in no render. I've just tried 1.12.4 and it still works. This:
`$j("#webhookPreviewBuildId").empty().append(myselect);`
is valid, but maybe something in the rendering or handling relied on the (otherwise extraneous) `.html()` call that was there previously:
`$j("#webhookPreviewBuildId").empty().append(myselect.html());`

Reply:
I think I have found it..
it's putting a `<select>` within the existing `<select id = "webhookPreviewBuildId">`, so there is two levels before the option items.

Reply:
maybe it needs to be `$j("#webhookPreviewBuildId").empty().append(myselect.val());`

Reply:
I assumed `#webhookPreviewBuildId` was a block-level element and not a `select` considering your existing code creates a select to begin with! Well, okay, I can fix this. ;)

Reply:
The `#webhookPreviewBuildId` is populated quite a bit in that method, and has change handlers and all there, so I don't want to recreate it each time.

Reply:
Also, if a user has previously selected a build from the list, it appears to stay selected if the same ID is in the list, which is a nice feature. Re-creating the `select` would (probably) reset it to the top of the select.

Reply:
Don't worry, what I'm planning on doing is just grabbing the existing select -- but no, your existing code _does_ empty and recreate the _contents_ of the select, so I'm guessing that the fact the select retains its selection is by virtue of the browser doing this for you.

Reply:
Yeah. That makes sense. 😃